packages/beacon-node/src/api/impl/validator/index.ts

```diff
@@ -553,8 +558,26 @@ export function getValidatorApi({
       const blockValueEngine = enginePayloadValue + gweiToWei(consensusBlockValueEngine); // Total block value is in wei
 
       let executionPayloadSource: ProducedBlockSource | null = null;
+      const shouldOverrideBuilder = fullBlock?.shouldOverrideBuilder ?? false;
+
+      // handle the builder override case separately
+      if (shouldOverrideBuilder === true) {
+        // this is just to make typescript happy as shouldOverrideBuilder can be true only will valid
+        // full block response
+        if (fullBlock === null) {
+          throw Error("Invalid null fullBlock with builder override");
+        }
```

Comment:
This should not be required as you are not accessing properties of `fullBlock` here

Reply:
probably correct, since there is a null check down the line as well for selected block